embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/SchemaType.kt

```diff
@@ -0,0 +1,44 @@
+package io.embrace.android.embracesdk.arch.schema
+
+import io.embrace.android.embracesdk.internal.utils.toNonNullMap
+import io.embrace.android.embracesdk.payload.AppExitInfoData
+
+internal sealed class SchemaType(
```

Comment:
Doesn't this map one-to-one with TelemetryType.subtype? And if so, we can roll these two together?

Reply:
My understand is that `emb.type` was not unique - it's a namespace. E.g. it would be possible to have N sources of data under `system.breadcrumb`. Is that not the case?

Reply:
The *schema* is unique per `subtype`, but we can potentially overload a particular one to mean different things depending on the data source. 

For instance, for if the `system.breadcrumb` subtype that has a single message field as the schema, we can have a data source for `CustomBreadcrumbs` that use it - or we can use it to expose a new breadcrumb for, I don't know, "new thread spawned", which just reuses that schema.

Customers might think of that as a new feature (since we are tracking new threads being created), but it's using the same base subtype so we don't have to roll another one. As long as it makes sense for the backend and UI to treat these the same way, there's no need to create a new subtype.

In the future, we might want to create another distinction so the UI can display the same subtype differently, but we can get there when we get there if creating a new subtype all the time for UI diffs isn't viable

Reply:
A good example is actually the Fragment Breadcrumb and a View Breadcrumb. Even if we offer two API methods to do each, the data that backs it is basically the same, so we can reuse the type.subtype (`ux.view`) but have two services be responsible for the capturing.

(Implementation wise, this example could probably be one service if we want, but you can see how services that eventually write to the same object defined by the same schema can/should live in separate services with vastly differently internal logic)